### DISCUSSION TITLE
Impossible to save required boolean fields with value of false

### DISCUSSION THREAD

**Discussion by mfalkvidd** [pods-framework/pods#6610] [Ideas]
commit 1e7dd942b (#1142) made it impossible to save a Required boolean field with the value false (the value must always be set to true).

How to reproduce:
1. Create a new pod.
2. Add a Yes / No field and check "Required" checkbox
3. Save the Pod
4. Add a new pod of the type you just created. Make sure the Yes/No value is unchecked.
5. Click preview.

Does "Required" make sense for a checkbox? Maybe the fix is to remove the "Required" option for boolean fields? But users might want to require the checkbox to be checked, for example when accepting a license agreement. For that case, the current implementation is good.

Possible workaround: Don't set fields of types boolean to Required in your pods. Maybe this should be the documented and recommended way.

**Comment by sc0ttkclark** [comment]
The required makes sense, but not if you're using a radio / dropdown select, it's meant for "you have to choose yes" type of situations. So, when you're using a radio, you assume that choosing _any_ value will result in a value that passes requirement, but boolean requirements are only passed if the "Yes" option is chosen.

**Comment by sc0ttkclark** [comment]
I think we could feasibly show an additional option when dropdown/select are chosen for input, called "Allow either value for Required validation" or similar text.

**Comment by jimtrue** [comment]
This would only be an 'enhancement'; you choose the type of field and require a selection.

- **Checkbox** would require that it is checked: Useful for Accept Terms & Conditions
- **Radio** would at least require something is selected, required will always work on this one, so this is good for defaults and just letting the user change the default values.
- **Drop-down** is ideal for those where you don't know what value they're going to choose but you definitely want them to choose something.  What's missing here is the 'no selection' choice that is typical in a select drop-down. That's the only enhancement that I could see working out of this issue request.

Thoughts?

**Comment by JoryHogeveen** [comment]
This was fixed in #5550

@jimtrue You tagged it as Documentation, still valid or can I close this?
